libnavigation-core/src/main/java/com/mapbox/navigation/core/routealternatives/RouteAlternativesCacheManager.kt

```diff
@@ -21,14 +22,24 @@ internal class RouteAlternativesCacheManager(
         }
     }
 
+    @VisibleForTesting
+    internal companion object {
+        internal const val CACHE_SIZE = 3
+    }
+
     init {
         navigationSession.registerNavigationSessionStateObserver(sessionStateObserver)
     }
 
     fun push(alternatives: List<DirectionsRoute>) {
-        cachedAlternatives.addAll(alternatives)
+        if (cachedAlternatives.size == CACHE_SIZE) {
+            cachedAlternatives.remove()
+        }
+        cachedAlternatives.add(alternatives)
```

Comment:
You use `ConcurrentLinkedQueue` but this method doesn't seem like a thread safe. Can `RouteAlternativesCacheManager` be accessed from multiple threads?

Reply:
nah, the method is consumed by router callback only